### ISSUE TITLE
Installing extras fails

### ISSUE THREAD

**Issue by nsuomine** [modxcms/revolution#16123]
### Summary
Installing extras does not anymore work after upgraded to MODX3. Not sure if this is a bug or something with my installation.

Getting error messages:

> Could not load class: modTransportPackage from mysql.modtransportpackage
> Could not get table name for class:
> Error 42000 executing statement: Array ( [0] => 42000 [1] => 1064 [2] => You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'AS `latestPackage` WHERE `latestPackage`.`package_name` = `modTra...' at line 3 )

**Comment by halftrainedharry** [comment]
Does this happen for all installations or only for a specific package?

If had the same error message in the 'installation console' when I tried to install `simplesearch-2.1.2-pl`. (Although the extra still seems to install correctly.)

But I didn't get the error for other packackes (like _Ace_ or _getResources_)

**Comment by nsuomine** [comment]
It does not happen with all packages but I got the same error with 2 different packages.  Both of them do not work after the installations even though Installer claims installations to be successful despite the error messages.

**Comment by halftrainedharry** [comment]
With `simplesearch-2.1.2-pl` the problem seems to be that the code in the resolver is not correct.

`$modx->getTableName('modTransportPackage')` in this line
https://github.com/Sterc/SimpleSearch/blob/fe709df019c3447733057e57d8940a95573b96d8/_build/resolvers/stercextra.resolver.php#L12
probably should be `$modx->getTableName('transport.modTransportPackage')`.
(As this resolver only sends some statistical data to the creator, it doesn't affect the installation.)

In short: It's probably not a MODX problem, but an error in the installation package.

---

Can you share the packages that didn't work for you?

**Comment by nsuomine** [comment]
Batcher and imageCropper failed.

**Comment by halftrainedharry** [comment]
All these extras are from the same creator (sterc), so they have the same error in the same resolver.

But as far as I can tell, _Batcher_ and _imageCropper_ install fine otherwise. They are probably just not compatible with MODX3.

**Comment by nsuomine** [comment]
Thanks for the information.

Batcher gives an error "Requested processor not found". Probably just an incompatibility issue.

I can't find ImageCropper anywhere in the menus so probably also an incompatibility issue.

**Comment by smg6511** [comment]
Closing, as these issues, if still existing, need to be reported at the individual Extra creators' repos.
